infra/network/aks-subnet-nsg.bicep

```diff
@@ -0,0 +1,34 @@
+@description('Required. The Network Security Group Name.')
+param name string
+
+@description('Required. The Azure region where the resources will be deployed.')
+param location string
+@description('Required. Environment name.')
+param environment string
+@description('Optional. Date in the format yyyy-MM-dd.')
+param createdDate string = utcNow('yyyy-MM-dd')
+@description('Optional. Date in the format yyyyMMdd-HHmmss.')
+param deploymentDate string = utcNow('yyyyMMdd-HHmmss')
+
+@description('Required. Array of Security Rules to deploy to the Network Security Group. When not provided, an NSG including only the built-in roles will be deployed.')
+param securityRules array = []
+
+var commonTags = {
+  Location: location
+  CreatedDate: createdDate
+  Environment: environment
+  Purpose: 'ADP-NSG'
```

Comment:
Purpose: ADP-NSG - AKS Subnet

Reply:
Actually - Shouldn't we have a generic NSG template that just takes a list of configuration so we can have many NSGs etc? @ppotturi  @ajaysoni-cap  @kbabigumira

Reply:
Updated the code according to our discussion this morning.